### PR TITLE
allow caching avatars for 7 days, allow overwriting default headers, remove IE8 hacks

### PR DESCRIPTION
**Pull request by butonic**
## Description
- allow caching avatars for 7 days by sending a `Cache-Control: max-age=604800` header. Will change for the user immedeately after changing his avatar. Can still be refreshed by pressing F5
- allow overwriting Cache-Control and Content-Type 
- remove IE8 hacks. OC no longer supports it.

## Related Issue

https://github.com/owncloud/core/issues/26585

## Motivation and Context

Performance, technical debt

## How Has This Been Tested?

Firefox and Edge work correctly, Chromium works as expected via http but not via https. I expect that to be a bug in chromium.

note that you might want to set `session.cache_limiter =`  to an empty value in your php.ini to prevent PHP from sending deprecated `Pragma: no-cache` headers that may interfere with IE (Firefox seems to ignore them)

## Screenshots (if appropriate):
unbenannt

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Techncal Debt

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by butonic** [comment]
maybe @felixheidecke has opinion ho long to cache the avatar?

File: core/Controller/AvatarController.php [hunk 1]
```diff
@@ -134,8 +134,7 @@ public function getAvatar($userId, $size) {
 			]);
 		}
 
-		$resp->addHeader('Pragma', 'public');
-		$resp->cacheFor(0);
+		$resp->cacheFor(604800); // allow caching for a week
```
Comment (PhilippSchaffrath) [inline core/Controller/AvatarController.php]: Can we go with 1 or 2 hours here? 7 days seems too long for me.

**Comment by DeepDiver1975** [comment]
During my local testing the request still hits the server. Is this intended?

**Comment by butonic** [comment]
@DeepDiver1975 The initial request, yes. Also, don't use F5 but just hit enter in the location bar. @PhilippSchaffrath and I also came up with a patch for disabling js triggered requests to preview.png. PR pending.

**Comment by PhilippSchaffrath** [comment]
@DeepDiver1975 can you make sure that you don't hit F5 during your test, this would re-load the avatar no matter what the header is. (like @butonic said, click in the address-bar and hit return)

The patch mentioned by @butonic is not really related to this issue.

**Comment by PVince81** [comment]
How to move forward ? Maybe 1 day is better.

My worry is mostly about invalidating. If someone changed their avatar, maybe it's fine if other users still see the old avatar for a day or so. But if the user who changed their avatar still sees the old avatar for a day, they'll think there's a bug and that the change didn't work.

**Comment by PVince81** [comment]
finish or close ?

**Comment by PVince81** [comment]
PR too old, there doesn't seem to be any interest in finishing it. Closing

**Pull request closed without merging**